packages/react-components/src/components/Tab/Tab.tsx

```diff
@@ -3,47 +3,62 @@ import cx from 'clsx';
 import { Text } from '../Typography';
 
 import styles from './Tab.module.scss';
+import { Badge } from '../Badge';
+import { Size } from 'utils';
 
 type HTMLProps =
   | (React.AnchorHTMLAttributes<HTMLAnchorElement> & { href: string })
   | (React.ButtonHTMLAttributes<HTMLButtonElement> & { href?: never });
 
 export type TabProps = HTMLProps & {
-  description?: React.ReactNode;
+  count?: number;
   isSelected?: boolean;
+  asBadge?: boolean;
+  size?: Size;
 };
 
 const baseClass = 'tab';
 
 export const Tab: React.FC<TabProps> = ({
   children,
   className,
-  description,
+  count,
   isSelected,
+  asBadge,
+  size = 'medium',
   ...restProps
 }) => {
+  const { disabled } =
+    restProps as React.ButtonHTMLAttributes<HTMLButtonElement>;
+
   return (
     <Text
       {...restProps}
       as={restProps.href ? 'a' : 'button'}
       size="md"
       bold={isSelected}
       className={cx(
-        styles[baseClass],
         className,
-        isSelected && styles[`${baseClass}--selected`]
+        styles[baseClass],
+        styles[`${baseClass}--${size}`],
+        isSelected && styles[`${baseClass}--selected`],
+        disabled && styles[`${baseClass}--disabled`]
       )}
     >
       {children}
-      {description && (
-        <Text
-          as="span"
-          size="md"
-          className={styles[`${baseClass}__description`]}
-        >
-          ({description})
+      {count && !asBadge && (
```

Comment:
I would consider extracting this condition and the next one outside the return statement.